ecc_gui.py

```diff
@@ -683,72 +683,54 @@ def update_initialize_population_button(self):
         """ Updates the Initialize Population button, by checking if all necessary files and settings are correct. If
             not, shows in the button what items are missing and makes the button do nothing. If criteria are met, button
             color is changed to green and button functionality is restored. """
-        isok, missing = self.can_generate_new_population()
-        if isok:
+        missing_messages = self.can_generate_new_population()
+        if len(missing_messages) == 0:
             self.generatechild.configure(relief="raised",
                                          bg="lightgreen", font=(DEFAULT_FONT, 12, "bold"), text="Initialize Population",
                                          width=52, height=6,
                                          state='normal',
                                          command=lambda: self.generate_next_population(self.settings['method']))
         else:
-            txt = ("Cannot Initialize Population:\n" + missing).rstrip()
+            messages = '\n'.join(missing_messages)
+            txt = f'Cannot Initialize Population:\n{messages}'
             self.generatechild.configure(relief=tk.RAISED, bg="#D0D0D0",
                                          font=(DEFAULT_FONT, 12, "bold"),
                                          width=52, height=6,
                                          activebackground="#D0D0D0",
                                          text=txt,
                                          state='disabled')
 
+    def is_setting_valid(self, setting_name):
+        return setting_name in self.settings and len(self.settings[setting_name]) != 0
+
     def can_generate_new_population(self):
         """ Function which checks all necessary files and settings for generating a new population. Returns a
-            False and string 'missing' with all missing features, or True if all criteria are met. """
-        missing = ""
-
-        if 'VAM base dir' in self.settings:
-            if len(self.settings['VAM base dir']) > 0:
-                pass
-            else:
-                missing += "· Please select the VAM base folder\n"
-        else:
-            missing += "· Please select the VAM base folder\n"
-
-        if 'appearance dir' in self.settings:
-            if len(self.settings['appearance dir']) > 0:
-                pass
-            else:
-                missing += "· Please select an Appearance folder\n"
+            False and message list with all missing features, or True if all criteria are met. """
+        messages = list()
+        if not self.is_setting_valid('VAM base dir'):
+            messages.append('· Please select the VAM base folder')
+        if not self.is_setting_valid('appearance dir'):
+            messages.append('· Please select an Appearance folder')
+        if not self.is_setting_valid('appearance dir'):

```

Comment:
line 714 needs to become:
`if not self.is_setting_valid('child template'):`

Reply:
I see, thanks!

Reply:
Fixed it.